components/home_page_search_discovery/home_page_search_discovery.react.js

```diff
@@ -0,0 +1,366 @@
+import React, { Component } from 'react';
+import Wrapper from '../global/wrapper/wrapper.react';
+import Form from '../global/form/form.react';
+import Input from '../global/form/input.react';
+import Select from '../global/form/select.react';
+import Button from '../global/button/button.react';
+import Link from '../global/link/link.react';
+import ErrorMessage from './child_components/errorMessage';
+import Data from './home_page_search_discovery_data.json';
+import './home_page_search_discovery.scss';
+
+class HomePageSearchDiscovery extends Component {
+  // Set properties
+  objGTM = {};
+
+  // Set the initial State
+  state = {
+    Data,
+    valueShow: '',
+    valueBetween: '',
+    valueAnd: '',
+    valueAcross: 'all',
+    errorShow: '',
+    errorBetween: '',
+    errorAnd: '',
+    success: null,
+    regEx: /^((0[1-9]|[[1-2][0-9]|3[0-1])\/(0[1-9]|1[012])\/)?[12][0-9]{3}$/
+  };
+
+  // Preserve the initial state in a new object
+  // so it can be reused
+  mainState = this.state;
+
+  // Create a unique DOM reference
+  // for hidden <input name="_dhs" value="y" />
+  hiddenRef = React.createRef();
+
+  /**
+   * ==================== IMPORTANT ========================
+   * All Class Methods auto bind using arrow function method
+   */
+
+  // Check input value on change
+  onChangeInput = e => {
+    if (e.target.id === 'search-all-collections') {
+      this.setState({ valueShow: e.target.value });
+    }
+    if (e.target.id === 'start-date') {
+      this.setState({ valueBetween: e.target.value });
+    }
+    if (e.target.id === 'end-date') {
+      this.setState({ valueAnd: e.target.value });
+    }
+  };
+
+  // Set the state to the initial state and hide the message
+  hideErrorMsg = (field, objKey, errMsg) => {
+    field !== '' ? this.setState({ [objKey]: errMsg }) : '';
+  };
+
+  // Check if field Show is empty or has a space inside
+  // Update the state on errorShow with the error Message
+  showErrorMsgIfEmptyOrWhiteSpace = e => {
+    return (field, objKey, errMsg) => {
+      if (field === ' ' || field === '') {
```

Comment:
Could we possibly change the condition to
`field.trim() === ''`?

This will avoid using `||` and will counter for any amount of whitespace.